Lib/test/test_unparse.py

```diff
@@ -130,7 +130,32 @@ class Foo: pass
 
 class ASTTestCase(unittest.TestCase):
     def assertASTEqual(self, ast1, ast2):
-        self.assertEqual(ast.dump(ast1), ast.dump(ast2))
+        # Ensure the comparisons start at an AST node
+        self.assertIsInstance(ast1, ast.AST)
+        self.assertIsInstance(ast2, ast.AST)
+
+        # An AST comparison routine modeled after ast.dump(), but
+        # instead of string building, it traverses the two trees
+        # in lock-step.  Defining an inner function also prevents
+        # the `missing` value from polluting the "real" function
+        # signature or module namespace.
```

Comment:
You can make the "missing" value a default argument to achieve the same, and the load will be slightly faster than the DEREF. The function is not public so is not a problem

Reply:
In the same commit, I moved `missing` to an argument default of the inner function.  I added the detail per Victor's request, but I'd gladly strike it just as well.